r: all CI pipelines failing

e.g. https://github.com/apache/arrow-adbc/actions/runs/5278455535/jobs/9547819529

```
── R CMD build ─────────────────────────────────────────────────────────────────
Error in rcb_init(self, private, super, path, dest_path, binary, vignettes,  : 
  argument "quiet" is missing, with no default
Calls: <Anonymous> ... <Anonymous> -> initialize -> rcb_init -> build_setup
Execution halted
Error: Process completed with exit code 1.
```

--- a/.github/workflows/native-unix.yml
+++ b/.github/workflows/native-unix.yml
@@ -639,7 +639,7 @@ jobs:
 
       - uses: r-lib/actions/setup-r-dependencies@v2
         with:
-          extra-packages: any::rcmdcheck, local::../adbcdrivermanager
+          extra-packages: any::rcmdcheck, local::../adbcdrivermanager, github::r-lib/pkgbuild@v1.4.0
           needs: check
           working-directory: r/${{ matrix.config.pkg }}
 

--- a/.github/workflows/native-unix.yml
+++ b/.github/workflows/native-unix.yml
@@ -639,6 +639,7 @@ jobs:
 
       - uses: r-lib/actions/setup-r-dependencies@v2
         with:
+          # Pin of pkgbuild can be removed after version 1.4.2 is released
           extra-packages: any::rcmdcheck, local::../adbcdrivermanager, github::r-lib/pkgbuild@v1.4.0
           needs: check
           working-directory: r/${{ matrix.config.pkg }}

